From 54a064291cf160b404a697d0b4b77951ec158602 Mon Sep 17 00:00:00 2001
From: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
Date: Tue, 27 Feb 2024 14:53:49 -0300
Subject: [PATCH] csi-powerstore: add label app.starlingx.io/component

This patch consists of changing the strategy from the Deployment resource and
add the app.starlingx.io/component label in all pods.
The strategy was needed changing because in the reapply the old replicasets
were not being deleted. The default value was rollingUpdate.
In this case, it doesn't effect functionality.

Signed-off-by: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
---
 charts/csi-powerstore/templates/controller.yaml | 3 +++
 charts/csi-powerstore/templates/node.yaml       | 1 +
 2 files changed, 4 insertions(+)

diff --git a/charts/csi-powerstore/templates/controller.yaml b/charts/csi-powerstore/templates/controller.yaml
index e827b7f..27a5c5d 100644
--- a/charts/csi-powerstore/templates/controller.yaml
+++ b/charts/csi-powerstore/templates/controller.yaml
@@ -171,10 +171,13 @@ spec:
   {{- else }}
   replicas: {{ required "Must provide the number of controller instances to create." .Values.controller.controllerCount }}
   {{- end }}
+  strategy:
+    type: Recreate
   template:
     metadata:
       labels:
         name: {{ .Release.Name }}-controller
+        app.starlingx.io/component: {{ index .Values "app.starlingx.io/component" }}
       annotations:
         kubectl.kubernetes.io/default-container: driver
     spec:
diff --git a/charts/csi-powerstore/templates/node.yaml b/charts/csi-powerstore/templates/node.yaml
index 04bb072..05a7572 100644
--- a/charts/csi-powerstore/templates/node.yaml
+++ b/charts/csi-powerstore/templates/node.yaml
@@ -97,6 +97,7 @@ spec:
     metadata:
       labels:
         app: {{ .Release.Name }}-node
+        app.starlingx.io/component: {{ index .Values "app.starlingx.io/component" }}
       {{- if or .Values.podmon.enabled .Values.csm.resiliency }}
         driver.dellemc.com: dell-storage
       {{- end }}
-- 
2.34.1
